commons/src/main/java/com/jagrosh/jdautilities/commons/waiter/EventWaiter.java

```diff
@@ -237,12 +236,10 @@ public final void onEvent(GenericEvent event)
             if(waitingEvents.containsKey(c))
             {
                 Set<WaitingEvent> set = waitingEvents.get(c);
-                WaitingEvent[] toRemove = set.toArray(new WaitingEvent[set.size()]);
 
                 // WaitingEvent#attempt invocations that return true have passed their condition tests
-                // and executed the action. We filter the ones that return false out of the toRemove and
-                // remove them all from the set.
-                set.removeAll(Stream.of(toRemove).filter(i -> i.attempt(event)).collect(Collectors.toSet()));
+                // and executed the action. We remove the ones that have successfully ran (those that returns true)
```

Comment:
This seems like the opposite behavior.
"We filter the ones that return false out of the toRemove and remove them all from the set." versus "We remove the ones that have successfully ran (those that returns true)"